File: Graph/GFG/Easy/AdjacencyListBFS/AdjacencyListBFS.java
// https://www.geeksforgeeks.org/problems/bfs-traversal-of-graph/1
import java.util.ArrayDeque;
import java.util.ArrayList;
import java.util.Deque;

public class AdjacencyListBFS {
    public ArrayList<Integer> bfsOfGraph(int V, ArrayList<ArrayList<Integer>> adj) {
        // Code here
        ArrayList<Integer> bfs = new ArrayList<>();
        int visited[] = new int[V];
        Deque<Integer> q = new ArrayDeque<>();
        q.offer(0);
        visited[0] = 1;
        bfs.add(0);
        while (!q.isEmpty()) {
            int curr = q.poll();
            for (int neighbor : adj.get(curr)) {
                if (visited[neighbor] == 0) {
                    q.offer(neighbor);
                    bfs.add(neighbor);
                    visited[neighbor] = 1;
                }
            }
        }
        return bfs;
    }
    public static void main(String[] args) {
        
    }
}

File: Graph/GFG/Easy/AdjacencyListBFS/AdjacencyListBFS.cpp
// https://www.geeksforgeeks.org/problems/bfs-traversal-of-graph/1
#include<bits/stdc++.h>
using namespace std;
vector<int> bfsOfGraph(int V, vector<int> adj[]) {
    queue<int> q;
    vector<int> visited(V, 0);
    vector<int> bfs;
    q.push(0);
    visited[0] = 1;
    while(!q.empty()) {
        int curr = q.front();
        bfs.push_back(curr);
        q.pop();
        for(auto neighbor: adj[curr]) {
            if(visited[neighbor] == 0) {
                q.push(neighbor);
                visited[neighbor] = 1;
            }
        }
    }
    return bfs;
}
int main(void) {
    
    return 0;
}